code/game/objects/items/weapons/explosives.dm

```diff
@@ -1,6 +1,6 @@
 //In this file: C4 and Syndicate Bombs
 /atom/
-	var/has_c4 = FALSE //If the item has c4 planted on it.
+	var/obj/item/weapon/c4/has_c4 = null //If the item has c4 planted on it.
```

Comment:
contents list is going to be an even uglier hack so I'd rather have it stay this way for now